addons/point_of_sale/static/tests/pos/tours/shipping_date_tour.js

```diff
@@ -0,0 +1,55 @@
+import * as Chrome from "@point_of_sale/../tests/pos/tours/utils/chrome_util";
+import * as Dialog from "@point_of_sale/../tests/generic_helpers/dialog_util";
+import * as ProductScreen from "@point_of_sale/../tests/pos/tours/utils/product_screen_util";
+import { registry } from "@web/core/registry";
+
+registry.category("web_tour.tours").add("ShippingDateTour", {
+    steps: () =>
+        [
+            ProductScreen.setTimeZone("America/New_York"),
+            Chrome.startPoS(),
+            Dialog.confirm("Open Register"),
+            ProductScreen.addOrderline("Whiteboard Pen", "1"),
+            ProductScreen.clickPayButton(),
+            {
+                content: "click ship later button",
+                trigger: ".button:contains('Ship Later')",
+                run: "click",
+            },
+            {
+                content: "pick a date",
+                trigger: '.modal-body input[type="date"]',
+                run: function () {
+                    const input = document.querySelector('.modal-body input[type="date"]');
+                    const nextYear = new Date().getFullYear() + 1;
+                    input.value = `${nextYear}-05-30`;
+                    input.dispatchEvent(new Event("input", { bubbles: true }));
+                    input.dispatchEvent(new Event("change", { bubbles: true }));
+                },
+            },
+            {
+                content: "click confirm button",
+                trigger: ".btn:contains('Confirm')",
+                run: "click",
+            },
+            {
+                content: "Assert shipping date was set",
+                trigger: ".payment-buttons .d-flex .btn span",
```

Comment:
If you use freezeDateTime. Here you should be able to assert that the date is set correctly inside the trigger with no need to do it in the run.